mlflow/spark.py

```diff
@@ -502,6 +529,7 @@ def _save_model_metadata(
         pyspark_version=pyspark.__version__,
         model_data=_SPARK_MODEL_PATH_SUB,
         code=code_dir_subpath,
+        model_class=_get_fully_qualified_class_name(spark_model),
```

Comment:
Does this work with Pipeline objects? Is there a need to handle dependency validation if the Pipeline contains an Estimator and a model Transformer object? (to include both scikit-learn and torch, for instance)

Reply:
Ah, I see... we're including both always. Got it!

Reply:
Do we want to do that if someone is just saving a feature transformer as a model?

Reply:
> Do we want to do that if someone is just saving a feature transformer as a model?

I think this is a rare case , for such case, if user want to reduce dependency package list, he can still set `pip_requirements` to override default pip requirements. WDYT ?

Reply:
I'm OK with this :) it doesn't seem worth the added complexity to handle that inspection just so that someone doesn't get pytorch installed with their encoding model.